### ISSUE TITLE
bcachefs BUG while reading in RO mode

### ISSUE THREAD

**Issue by jpsollie** [koverstreet/bcachefs#162]
While performing emergency backup (not sure about state of background target any longer, I wanted to backup everything before starting to mess around with underlying medium), so I mounted the whole set RO, but bcachefs doesn't seem to like it:
```
BUG: kernel NULL pointer dereference, address: 0000000000000021
#PF: supervisor read access in kernel mode
#PF: error_code(0x0000) - not-present page
PGD 0 P4D 0
Oops: 0000 [#1] SMP
CPU: 5 PID: 4694 Comm: rsync Tainted: G        W         5.9.10+ #9
Hardware name: Gigabyte Technology Co., Ltd. X399 DESIGNARE EX/X399 DESIGNARE EX-CF, BIOS F12 12/11/2019
RIP: 0010:kmem_cache_alloc+0xed/0x380
Code: 65 4c 03 05 8d cd e3 7e 49 83 78 10 00 4d 8b 20 0f 84 62 02 00 00 4d 85 e4 0f 84 59 02 00 00 8b 45 28 48 8b 7d 00 48 8d 4a 01 <49> 8b 1c 04 4c 89 e0 65 48 0f c7 0f 0f 94 c0 84 c0 74 ba 8b 4>
RSP: 0018:ffff888770f4b960 EFLAGS: 00010202
RAX: 0000000000000020 RBX: ffffea00095e5b40 RCX: 000000000022c9bf
RDX: 000000000022c9be RSI: 0000000000008d40 RDI: 00000000000220a0
RBP: ffff88885b443b00 R08: ffff88885e9620a0 R09: ffff888805388440
R10: 0000000000000000 R11: 0000000000000000 R12: 0000000000000001
R13: 0000000000000000 R14: 0000000000008d40 R15: ffff8886363faf00
FS:  00007fa7318fb740(0000) GS:ffff88885e940000(0000) knlGS:0000000000000000
CS:  0010 DS: 0000 ES: 0000 CR0: 0000000080050033
CR2: 0000000000000021 CR3: 0000000803db7000 CR4: 00000000001506e0
Call Trace:
 __bch2_page_state_create+0x16/0x40 [bcachefs]
 bch2_readahead+0x4e6/0x750 [bcachefs]
 ? add_to_page_cache_vec+0x21c/0x440
 ? scan_shadow_nodes+0x30/0x30
 read_pages+0x5f/0x240
 page_cache_readahead_unbounded+0x13f/0x1e0
 generic_file_buffered_read+0x280/0xc50
 ? ttwu_queue_wakelist+0xca/0xe0
 ? try_to_wake_up+0x171/0x280
 ? tty_insert_flip_string_fixed_flag+0x8f/0xf0
 bch2_read_iter+0x11f/0x3d0 [bcachefs]
 ? pty_write+0x9d/0xa0
 ? remove_wait_queue+0x18/0x60
 ? __wake_up_common_lock+0x75/0x90
 new_sync_read+0xf9/0x170
 vfs_read+0xc8/0x150
 ksys_read+0x5b/0xe0
 do_syscall_64+0x2d/0x40
 entry_SYSCALL_64_after_hwframe+0x44/0xa9
RIP: 0033:0x7fa731a2616e
Code: c0 e9 c6 fe ff ff 48 8d 3d 2f c0 06 00 50 e8 a9 cb 01 00 66 0f 1f 84 00 00 00 00 00 64 8b 04 25 18 00 00 00 85 c0 75 14 0f 05 <48> 3d 00 f0 ff ff 77 5a c3 66 0f 1f 84 00 00 00 00 00 48 83 e>
RSP: 002b:00007ffdef7d68f8 EFLAGS: 00000246 ORIG_RAX: 0000000000000000
RAX: ffffffffffffffda RBX: 000055640ae935c0 RCX: 00007fa731a2616e
RDX: 0000000000040000 RSI: 000055640ae93c50 RDI: 0000000000000003
RBP: 0000000000040000 R08: 0000000000000000 R09: 000055640ae93c50
R10: 000000000000001c R11: 0000000000000246 R12: 0000000000000000
R13: 0000000000000000 R14: 0000000000000000 R15: 0000000000040000
Modules linked in: xfs bcachefs lz4_compress crc64 lz4_decompress btrfs blake2b_generic xor lzo_compress lzo_decompress mq_deadline raid6_pq ses enclosure input_leds iwlmvm it87 led_class hwmon_v>
CR2: 0000000000000021
---[ end trace f5349d3f484ff7b4 ]---
RIP: 0010:kmem_cache_alloc+0xed/0x380
Code: 65 4c 03 05 8d cd e3 7e 49 83 78 10 00 4d 8b 20 0f 84 62 02 00 00 4d 85 e4 0f 84 59 02 00 00 8b 45 28 48 8b 7d 00 48 8d 4a 01 <49> 8b 1c 04 4c 89 e0 65 48 0f c7 0f 0f 94 c0 84 c0 74 ba 8b 4>
RSP: 0018:ffff888770f4b960 EFLAGS: 00010202
RAX: 0000000000000020 RBX: ffffea00095e5b40 RCX: 000000000022c9bf
RDX: 000000000022c9be RSI: 0000000000008d40 RDI: 00000000000220a0
RBP: ffff88885b443b00 R08: ffff88885e9620a0 R09: ffff888805388440
R10: 0000000000000000 R11: 0000000000000000 R12: 0000000000000001
R13: 0000000000000000 R14: 0000000000008d40 R15: ffff8886363faf00
FS:  00007fa7318fb740(0000) GS:ffff88885e940000(0000) knlGS:0000000000000000
CS:  0010 DS: 0000 ES: 0000 CR0: 0000000080050033
CR2: 0000000000000021 CR3: 0000000803db7000 CR4: 00000000001506e0
```
This error occurs when performing rsync to USB disk.
Leads to kernel panic, not sure whether this is the cause, but I found these in the logs after this BUG:
```
 BUG: kernel NULL pointer dereference, address: 0000000000000021
 #PF: supervisor read access in kernel mode
 #PF: error_code(0x0000) - not-present page
 PGD 0 P4D 0
 Oops: 0000 [#2] SMP
 CPU: 28 PID: 151 Comm: kworker/28:0 Tainted: G      D           5.9.10+ #9
 Hardware name: Gigabyte Technology Co., Ltd. X399 DESIGNARE EX/X399 DESIGNARE EX-CF, BIOS F12 12/11/2019
 Workqueue:  0x0 (events)
 RIP: 0010:kmem_cache_alloc+0xed/0x380
 Code: 65 4c 03 05 8d cd e3 7e 49 83 78 10 00 4d 8b 20 0f 84 62 02 00 00 4d 85 e4 0f 84 59 02 00 00 8b 45 28 48 8b 7d 00 48 8d 4a 01 <49> 8b 1c 04 4c 89 e0 65 48 0f c7 0f 0f 94 c0 84 c0 74 ba 8b 4>
 RSP: 0018:ffff88885af07d80 EFLAGS: 00010202
 RAX: 0000000000000020 RBX: ffff888859324e40 RCX: 000000000000058b
 RDX: 000000000000058a RSI: 0000000000000cc0 RDI: 00000000000220a0
 RBP: ffff88885b443b00 R08: ffff88885ef220a0 R09: 0000000000000000
 R10: ffff88885af07e88 R11: ffff88895af07e7b R12: 0000000000000001
 R13: 0000000000000000 R14: 0000000000000cc0 R15: ffffffff810b81d0
 FS:  0000000000000000(0000) GS:ffff88885ef00000(0000) knlGS:0000000000000000
 CS:  0010 DS: 0000 ES: 0000 CR0: 0000000080050033
 CR2: 0000000000000021 CR3: 000000065a441000 CR4: 00000000001506e0
 Call Trace:
  __kthread_create_on_node+0x70/0x1a0
  kthread_create_on_node+0x34/0x40
  ? kmem_cache_alloc+0x327/0x380
  create_worker+0xc7/0x1b0
  worker_thread+0x2c6/0x3c0
  ? alloc_unbound_pwq+0x300/0x300
  kthread+0xf2/0x130
  ? kthread_park+0x80/0x80
  ret_from_fork+0x1f/0x30
 Modules linked in: xfs bcachefs lz4_compress crc64 lz4_decompress btrfs blake2b_generic xor lzo_compress lzo_decompress raid6_pq mq_deadline ses enclosure input_leds iwlmvm hid_generic led_class >
 CR2: 0000000000000021
 ---[ end trace 7e5244f8842c28b5 ]---
 RIP: 0010:kmem_cache_alloc+0xed/0x380
 Code: 65 4c 03 05 8d cd e3 7e 49 83 78 10 00 4d 8b 20 0f 84 62 02 00 00 4d 85 e4 0f 84 59 02 00 00 8b 45 28 48 8b 7d 00 48 8d 4a 01 <49> 8b 1c 04 4c 89 e0 65 48 0f c7 0f 0f 94 c0 84 c0 74 ba 8b 4>
 RSP: 0018:ffff888661aef960 EFLAGS: 00010202
 RAX: 0000000000000020 RBX: ffffea0019644fc0 RCX: 000000000000058b
 RDX: 000000000000058a RSI: 0000000000008d40 RDI: 00000000000220a0
 RBP: ffff88885b443b00 R08: ffff88885ef220a0 R09: ffff88865941ab80
 R10: 0000000000000000 R11: 0000000000000000 R12: 0000000000000001
 R13: 0000000000000000 R14: 0000000000008d40 R15: ffff8888597b6b00
 FS:  0000000000000000(0000) GS:ffff88885ef00000(0000) knlGS:0000000000000000
 CS:  0010 DS: 0000 ES: 0000 CR0: 0000000080050033
 CR2: 0000000000000021 CR3: 000000065a441000 CR4: 00000000001506e0
```
mounting RW fixed the issue.

**Comment by koverstreet** [comment]
Fixed by 6f680b8640

**Comment by koverstreet** [comment]
Not sure about the bug in __kthread_create_on_node, that doesn't look bcachefs related
